torchao/prototype/autoround/core.py

```diff
@@ -0,0 +1,231 @@
+import os
+from dataclasses import dataclass
+from typing import Dict, Tuple
+
+import torch
+from torch.utils._pytree import tree_flatten, tree_unflatten
+
+import torchao.prototype.autoround.utils as ar_utils
+import torchao.quantization as ao_quant
+from torchao.dtypes import TensorCoreTiledLayoutType, to_affine_quantized_static
+from torchao.prototype.autoround.multi_tensor import MultiTensor
+from torchao.quantization.quant_primitives import ZeroPointDomain
+from torchao.utils import find_multiple
+
+# TODO: remove it before merge
+ar_utils.freeze_random()
+
+
+@dataclass
+class AutoRoundConfig:
+    bits: int = 4
+    sym: bool = False
+    iters: int = 200
+    group_size: int = 128
+    train_bs: int = 4
+    eval_bs: int = 4
+    seed: int = 42
+    amp: bool = True
+    nsamples: int = 128
+    seqlen: int = 2048
+    quant_lm_head: bool = False
+
+
+auto_round_config = AutoRoundConfig()
+
+
+@torch.no_grad()
+def create_qmodel_from_qdq_model(qdq_model: torch.nn.Module):
+    """Create a quantized model from a qdq model.
+
+    The qdq_model includes Linear quantized by auto-round, which includes qdq weight, scale, zp.
+    """
+
+    @torch.no_grad()
+    def apply_static_quant(observed_linear):
+        device = observed_linear.weight.device
+        scale = observed_linear.scale.to(device)
+        zero_point = observed_linear.zp.to(device)
+
+        def to_uintx_weight(input_float):
```

Comment:
Hi @jerryzh168, I followed the `uintx_weight_only` to create `AffineQuantizedTensor` for other bits. Please have a look. Thanks.

https://github.com/pytorch/ao/blob/b523f9f9e15b6fb80d10f585d9cf45e0c5e4d10e/torchao/quantization/quant_api.py#L471-L484